### PR TITLE
Removed url.QueryEscape from the make*Endpoint functions

### PR DESCRIPTION
**Pull request by LennartOlsen**
Simple example of the issue here:
https://play.golang.org/p/irnijFBj60

### CODE REVIEW

File: examples/profilesvc/transport.go [hunk 1]
```diff
@@ -227,63 +225,63 @@ func decodeDeleteAddressRequest(_ context.Context, r *http.Request) (request int
...
-	req.Method, req.URL.Path = "POST", url.QueryEscape("/profiles/")
+	req.Method, req.URL.Path = "POST", "/profiles/"
 	return encodeRequest(ctx, req, request)
 }
 
 func encodeGetProfileRequest(ctx context.Context, req *http.Request, request interface{}) error {
 	// r.Methods("GET").Path("/profiles/{id}")
 	r := request.(getProfileRequest)
-	req.Method, req.URL.Path = "GET", url.QueryEscape("/profiles/"+r.ID)
+	req.Method, req.URL.Path = "GET", "/profiles/"+r.ID
```
Comment (peterbourgon) [inline examples/profilesvc/transport.go]: Do we still want to QueryEscape the e.g. ID?

File: examples/profilesvc/transport.go [hunk 2]
```diff
@@ -227,63 +225,63 @@ func decodeDeleteAddressRequest(_ context.Context, r *http.Request) (request int
...
-	req.Method, req.URL.Path = "GET", url.QueryEscape("/profiles/"+r.ProfileID+"/addresses/")
+	req.Method, req.URL.Path = "GET", "/profiles/"+r.ProfileID+"/addresses/"
 	return encodeRequest(ctx, req, request)
 }
 
 func encodeGetAddressRequest(ctx context.Context, req *http.Request, request interface{}) error {
 	// r.Methods("GET").Path("/profiles/{id}/addresses/{addressID}")
 	r := request.(getAddressRequest)
-	req.Method, req.URL.Path = "GET", url.QueryEscape("/profiles/"+r.ProfileID+"/addresses/"+r.AddressID)
+	req.Method, req.URL.Path = "GET", "/profiles/"+r.ProfileID+"/addresses/"+r.AddressID
```
Comment (peterbourgon) [inline examples/profilesvc/transport.go]: Likewise the two IDs here? (Other comments elided, but you get the same issue with other methods.)

Comment (LennartOlsen) [inline examples/profilesvc/transport.go]: You are right - but if we are escaping ids here, we should also be escaping them when saving.
I get that its user input and that should generally be escaped.

IMHO i wouldn't escape the path itself, only the query string eg. whats after in the url "?"

Comment (peterbourgon) [inline examples/profilesvc/transport.go]: For example if an ID is `"foo bar"` then it should be stored with a space literal but in the URL as `foo%20bar`.

Comment (LennartOlsen) [inline examples/profilesvc/transport.go]: You are right @peterbourgon ill do a new commit

**Comment by peterbourgon** [comment]
Thank you!

**Pull request merged**